pkg/generate/app/cmd/newapp.go

```diff
@@ -916,16 +916,31 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	}, nil
 }
 
+func (c *AppConfig) findImageStreamInObjectList(objects app.Objects, name, namespace string) *imageapi.ImageStream {
+	for _, check := range objects {
+		if is, ok := check.(*imageapi.ImageStream); ok {
+			nsToCompare := is.Namespace
+			if len(nsToCompare) == 0 {
+				nsToCompare = c.OriginNamespace
+			}
+			if is.Name == name && nsToCompare == namespace {
+				return is
+			}
+		}
+	}
+	return nil
+}
+
 // crossStreamCircularTagReference inherits some logic from imageapi.FollowTagReference, but differs in that a) it is only concerned
 // with whether we can definitively say the IST chain is circular, and b) can cross image stream boundaries;
 // not in imageapi pkg (see imports above) like other helpers cause of import cycle with the image client
-func (c *AppConfig) crossStreamCircularTagReference(stream *imageapi.ImageStream, tag string) bool {
-	seen := sets.NewString()
-	// get our own copy
-	strm, err := c.ImageClient.ImageStreams(stream.ObjectMeta.Namespace).Get(stream.ObjectMeta.Name, metav1.GetOptions{})
-	if strm == nil || err != nil {
+func (c *AppConfig) crossStreamCircularTagReference(stream *imageapi.ImageStream, tag string, objects app.Objects) bool {
+	if stream == nil {
 		return false
 	}
+	seen := sets.NewString()
+	// copy pointer as we'll update it
+	strm := stream
```

Comment:
i don't understand the need for this copy.  Assigning stream=blah does not affect the value pointed to by stream.  (assigning *stream=blah would, of course).

Reply:
right - over thought it - will adjust